src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -237,7 +237,7 @@ private async Task ExecuteAsync(HttpContext context, ConnectionDelegate connecti
                     // Wait for the transport to run
                     // Ignore exceptions, it has been logged if there is one and the application has finished
                     // So there is no one to give the exception to
-                    await connection.TransportTask!.NoThrow();
+                    await ((Task)connection.TransportTask!).ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing);
```

Comment:
What is the cast for?

Reply:
`TransportTask` is `Task<bool>` and in general `Task<TResult>`s can't have `ConfigureAwaitOptions.SuppressThrowing` because of the result returning semantics.
Thus that task is casted to `Task` in order to specify the throwing-behavior.

Reply:
https://github.com/dotnet/runtime/blob/202ccdb59afedcf2d7e86b3ba10a2b9a703c71c9/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Future.cs#L536-L550

I see what is happening. Thanks 👍